test_conformance/c11_atomics/common.h

```diff
@@ -873,20 +872,23 @@ std::string CBasicTest<HostAtomicType, HostDataType>::KernelCode(cl_uint maxNumD
         "  if(get_local_id(0) == 0) // first thread in workgroup\n";
     else
       // global atomics declared in program scope
-      code +=
-      "  if(atomic_fetch_add(&finishedThreads, 1) == get_global_size(0)-1)\n"
-      "    // last finished thread\n";
+      code += "  if(atomic_fetch_add_explicit(&finishedThreads, 1) == "
+              "get_global_size(0)-1, memory_order_relaxed, "
+              "memory_scope_work_group)\n"
+              "    // last finished thread\n";
```

Comment:
The `memory_order_relaxed` and `memory_scope_work_group` look wrong here, they're free expressions not in the call?

```diff
-if(atomic_fetch_add_explicit(&finishedThreads, 1) == get_global_size(0)-1, memory_order_relaxed, memory_scope_work_group)
+if(atomic_fetch_add_explicit(&finishedThreads, 1, memory_order_relaxed, memory_scope_work_group) == get_global_size(0)-1)
```

If this didn't cause any `clBuildProgram` type errors is this code even traversed?

Reply:
Thinking out loud.

It looks like every thread calls `atomic_fetch_add_explicit` with an operand of `1` and the work-item that returns `get_global_size(0)-1` knows it was last so then performs some other action.  The `memory_scope_work_group` is only constraining the ordering of these operations, and `+1` on `finishedThreads` is order invariant, so the `memory_scope_work_group` operation on a global atomic here is just the same as the previous `memory_order_seq_cst`?

The only required response to this comment may be "Yes."

Reply:
Thanks, I already addressed the arguments missing the call, but yes I think we were not hitting this case when the kernel string was getting built.

Reply:
> Thinking out loud.
> 
> It looks like every thread calls `atomic_fetch_add_explicit` with an operand of `1` and the work-item that returns `get_global_size(0)-1` knows it was last so then performs some other action. The `memory_scope_work_group` is only constraining the ordering of these operations, and `+1` on `finishedThreads` is order invariant, so the `memory_scope_work_group` operation on a global atomic here is just the same as the previous `memory_order_seq_cst`?
> 
> The only required response to this comment may be "Yes."

Yes. This sounds reasonable to me. The changes shouldn't make any difference since the `+1` atomic operation is not dependent on order.

Reply:
Thanks.